irc-dcc 2.0.0 & xdcc 1.1.1

Ping @JanGe

Building xdcc-1.1.1...
Preprocessing executable 'xdcc' for xdcc-1.1.1...
[1 of 6] Compiling IRC.Types        ( src/IRC/Types.hs, dist/build/xdcc/xdcc-tmp/IRC/Types.o )
[2 of 6] Compiling ZNC              ( src/ZNC.hs, dist/build/xdcc/xdcc-tmp/ZNC.o )

src/ZNC.hs:18:1: warning: [-Wmissing-signatures]                                                                                [0/1971]
    Top-level binding with no type signature:
      ZncStatusUser :: forall t.
                       (Data.String.IsString
                          (Network.IRC.Conduit.Internal.Messages.NickName t),
                        Eq (Network.IRC.Conduit.Internal.Messages.NickName t)) =>
                       Source t

src/ZNC.hs:20:1: warning: [-Wmissing-signatures]
    Top-level binding with no type signature:
      ZncDisconnectedMsg :: forall a.
                            (Data.String.IsString a, Eq a) =>
                            Network.IRC.Conduit.Internal.Messages.Target a -> Message a
[3 of 6] Compiling DCC              ( src/DCC.hs, dist/build/xdcc/xdcc-tmp/DCC.o )
[4 of 6] Compiling XDCC             ( src/XDCC.hs, dist/build/xdcc/xdcc-tmp/XDCC.o )

src/XDCC.hs:48:17: error:
    • Couldn't match type ‘Stati’ with ‘()’
      Expected type: IRC.EventHandler Stati -> XdccIO ()
        Actual type: IRC.EventHandler () -> XdccIO ()
    • In the expression: XdccIO . IRC.addHandler
      In an equation for ‘addDccHandler’:
          addDccHandler = XdccIO . IRC.addHandler

src/XDCC.hs:48:26: error:
    • Couldn't match type ‘()’ with ‘Stati’
      Expected type: IRC.EventHandler () -> IRC.StatefulIRC Stati ()
        Actual type: IRC.EventHandler () -> IRC.StatefulIRC () ()
    • In the second argument of ‘(.)’, namely ‘IRC.addHandler’
      In the expression: XdccIO . IRC.addHandler
      In an equation for ‘addDccHandler’:
          addDccHandler = XdccIO . IRC.addHandler
curators@stk-build-i-2a9f06b4:~/stackage/automated$

--- a/build-constraints.yaml
+++ b/build-constraints.yaml
@@ -2484,10 +2484,6 @@ packages:
         # https://github.com/fpco/stackage/issues/1796
         - generic-deriving < 1.11
         - text-show < 3.4
-
-        # https://github.com/fpco/stackage/issues/1800
-        - irc-dcc < 2.0.0
-        - xdcc < 1.1.0
 # end of packages
 
 

